python/foglamp/plugins/south/http_south/http_south.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""HTTP Listener handler for sensor readings"""
+import sys
+from aiohttp import web
+import asyncio
+from foglamp.common import logger
+from foglamp.common.web import middleware
+from foglamp.services.south.ingest import Ingest
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_LOGGER = logger.setup(__name__)
+
+_CONFIG_CATEGORY_NAME = 'HTTP_SOUTH'
+_CONFIG_CATEGORY_DESCRIPTION = 'South Plugin HTTP Listener'
+_DEFAULT_CONFIG = {
+    'plugin': {
+         'description': 'http_south_device',
+         'type': 'string',
+         'default': 'http_south'
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '6683',
+    },
+    'host': {
+        'description': 'Address to accept data on',
+        'type': 'string',
+        'default': '0.0.0.0',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': 'sensor-reading',
+    }
+}
+
+
+def plugin_info():
+    return {'name': 'http_south', 'version': '1.0', 'mode': 'async', 'type': 'device',
+            'interface': '1.0', 'config': _DEFAULT_CONFIG}
+
+
+def plugin_init(config):
+    """Registers HTTP Listener handler to accept sensor readings"""
+
+    _LOGGER.info("Retrieve HTTP Listener Configuration %s", config)
+
+    host = config['host']['value']
+    port = config['port']['value']  # Make port dynamic? its unattended env! expose as service port?
```

Comment:
There is nothing to gain my making this dynamic as it is a port that has to be known by the sensors sending to it.

This is not the service port as the service port id the port for REST API services.